File: common/src/main/java/dev/obscuria/fragmentum/core/v1/common/signal/BaseSignal1.java
package dev.obscuria.fragmentum.core.v1.common.signal;

import dev.obscuria.fragmentum.api.v1.common.signal.Signal1;
import org.jetbrains.annotations.ApiStatus;

@ApiStatus.Internal
public final class BaseSignal1<P1> extends BaseSignal<Signal1.Listener<P1>> implements Signal1<P1>
{
    public void emit(P1 p1)
    {
        this.emit(listener -> listener.consume(p1));
    }
}

File: common/src/main/kotlin/dev/obscuria/fragmentum/core/tools/signal/BaseSignal1.kt
package dev.obscuria.fragmentum.core.tools.signal

import dev.obscuria.fragmentum.api.tools.signal.Signal1

class BaseSignal1<P1> : BaseSignal<Signal1.Listener<P1>>(), Signal1<P1>
{
    override fun emit(p1: P1) = emit { it.consume(p1) }
}